### ISSUE TITLE
Update state logic start date

### ISSUE THREAD

**Issue by mike-audi** [tiki/apps-receipt-capacitor#326]
### Reason

The 4 week payout period should start only ONCE the user has completed the gmail & retailer linking

### Description

Requirements:
- The clock for the 4 week period should only start once a user has connected both a gmail and a retailer account
- If one of the accounts becomes unverified or disconnected during the 4 week period, the user should have a 1 week buffer to fix it before the clock is reset. 
- The start should be reset when a user is issued a payout.

### Success Criteria

- [x] approve state logic changes
- [ ] implement start clock on successful connection
- [ ] implement clock reset on successful payout
- [ ] implement buffer window for "broken" connections

### Additional Information

Potential solution:
- Add a start date to the state service which can be used with the existing status functions to always create a forward looking window.
- Add disconnect date to the state service for handling the buffer period.

**First step (BEFORE ANY CODE): propose the state changes with pseudo code.**

**Comment by MiroBenicio** [comment]
State Sync: 
	Create a Start Date State
	Create a Disconnect Date State

Login Flow:
	When a login happens, check if the start date exists, if not, check the state accounts, if the other type of account its already connected, create a 
	new start date, if its not connected yet, do nothing
	if the date already exists, do nothing

Logout Flow:
	Check if its the unique account connect (already done for the UI flow)
	if the logout happens and its the only account that was connected, check the start date, if its not null, set the disconnect date state

Initialize Flow:
	Check the disconnect date state
	if its undefined and have a start date, add a date to the state sync
	if it have a disconnected date, check the accounts, if it has one disconnected/unverified at least, check the number of days since the disconnect date, if its greater or equal than 7
	reset everything, its less than 7, do nothing
	if it have 2 accounts connected, set the disconnect date to undefined

Payout:
	Reset everything

**Comment by mike-audi** [comment]
> if its undefined and have a start date, add a date to the state sync
> if it have a disconnected date, check the accounts, if it has one disconnected/unverified at least, check the number of days since the disconnect date, if its greater or equal than 7
> reset everything, its less than 7, do nothing

I'm not sure I understand this:
- if its undefined and have a start date, add a date to the state sync

For this: if it have a disconnected date, check the accounts, if it has one disconnected/unverified at least, check the number of days since the disconnect date, if its greater or equal than 7 reset everything, its less than 7, do nothing

it should be reset if there is < 1 verified gmail or < 1 verified retailer

**Comment by MiroBenicio** [comment]
if the disconnect date is undefined and have a start date, add a date to the state sync:

During the initialization, if the disconnect date is undefined and the start date exists, means the user have the 2 account types connected and verified, so we add a date into the sync state

"it should be reset if there is < 1 verified gmail or < 1 verified retailer":
if there is >= 1 verified gmail or >= 1 verified retailer the disconnected date will be undefined, so will not reset, so, if there is a disconnect date will be the same

**Comment by mike-audi** [comment]
@MiroBenicio this is good to go. Please start building :).
